rust/relay/src/main.rs

```diff
@@ -27,13 +29,46 @@ struct Args {
     /// Must not be a wildcard-address.
     #[arg(long, env)]
     listen_ip4_addr: Ipv4Addr,
+    /// The websocket URL of the portal server to connect to.
+    ///
+    /// If omitted, the relay server will start immediately, otherwise we first log on and wait for the `init` message.
+    #[arg(long, env)]
+    portal_ws_url: Option<Url>,
+    /// Token generated by the portal to authorize websocket connection
+    #[arg(long, env)]
+    portal_token: String,
+    /// Whether to allow connecting to the portal over an insecure connection.
+    #[arg(long)]
+    allow_insecure_ws: bool,
     /// A seed to use for all randomness operations.
     ///
-    /// Useful for testing and only available in debug builds.
+    /// Only available in debug builds.
     #[arg(long, env)]
     rng_seed: Option<u64>,
 }
 
+// TODO: Code repetition from common
+fn get_websocket_path(mut url: Url, token: String, ipv4: Ipv4Addr) -> Result<Url> {
```

Comment:
@thomaseizinger I wanted to test everything together in a single docker-compose so I based myself on your opened draft pr, I needed to add this

Reply:
I was expecting the user to pass in the correct WS url, something like: `wss://portal.firez.one/relay/websocket?token=XYZ`.

Does that not work for your usecase? In other words, the fact that we have an auth token is opaque to the our code.

Reply:
Hmm, I think it's okay but it might be a bit confusing as UI, specially since we do something like this for gateway/headless clients.

So we should use the same convention for both cases.

Reply:
> Hmm, I think it's okay but it might be a bit confusing as UI, specially since we do something like this for gateway/headless clients.
> 
> So we should use the same convention for all cases.

I am happy to go either way although I don't really understand what is better about splitting this up. Isn't it easier to configure for users if we give them a single URL to copy and configure vs two parameters?

Reply:
+1 it should be consistent across all binaries. I think the reason we had them separated was because we'll _also_ be using X.509s to authenticate Gateways (and Relays, and headless clients) to leverage an org's existing PKI.

This will come later though -- we can add another param/envvar to provide a base64-encoded X.509 when it comes time.

For now, I supposed embedding the token in the URL is simplest?

Reply:
If we keep the token in the URL it requires chaining  the ffi interface so I think the simplest is to keep the token as a separate parameter for now.

Reply:
I've opened a follow-up issue for this: https://github.com/firezone/product/issues/614